src/ui_components/search_input.js

```diff
@@ -21,24 +24,29 @@ export default class SearchInput {
 
   /* Singleton */
   static initSearchInput(tagSelector) {
-    if (! window.__searchInput) {
-      window.__searchInput = new SearchInput(tagSelector);
+    if (window.__searchInput) {
+      return window.__searchInput;
+    }
+
+    window.__searchInput = new SearchInput(tagSelector);
 
-      window.clearSearch = () => {
-        window.__searchInput.searchInputHandle.value = '';
-        window.app.navigateTo('/');
+    window.clearSearch = () => {
+      const isMobile = isMobileDevice();
+      window.__searchInput.searchInputHandle.value = '';
+      window.app.navigateTo('/');
+      if (!isMobile || isMobile && document.activeElement.id === SEARCH_INPUT_ID) {
```

Comment:
You don't need the `SEARCH_INPUT_ID` constant, the `tagSelector` parameter is already what you need.